### PR TITLE
[READY] Kitchen Rework part 1: I just want to griddle!

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus, Kryson
 add: Adds the griddle, a panless stove/grill for you to bake your meat products on. Nanotrasen has decided to move away from microwaving everything
 tweak: Burgers are now made from patties, patties are grilled on the griddle, raw patties are made by using a rolling pin on a raw meatball
 tweak: Most meat is grilled on the griddle now instead of microwaved
 tweak: More meatballs are gained from meat
 sprite: new meatball, patty and sausage sprites
/:cl:

This PR essentialy moves away from the extremely microwave dependent cooking we have for meat right now, and making it a bit more sensical by making you use a grill to grill meat. The grill takes a different time (with variation) for different grilled things. Once finished it will turn that food into something else.

Yes, this does mean creating burgers takes longer, but in return you can make more patties at once, and you are not required to stay at the grill while its going. This lets you cook as much as you want at once, just make sure your meat doesn't burn!

In the future, I hope to move more things like this to machines similar to this (Pasta boiling, putting eggs on the griddle, soup making, etcetera) to create for a more interesting cooking experience.

https://streamable.com/foozub

TODOS:
- [x] map in the griddles
- [x] bug test a bit
- [x] better/new audio for the grille
- [x] add smoke as overlay on cooked foods
- [x] imrpove flavor text for grilling

### CODE REVIEW

**Comment by BootlegBow** [comment]
dear god this is the best pr i've seen in a month

**Comment by KathrinBailey** [comment]
I like the sprites and heavily appreciate moving towards making the kitchen more like a kitchen.
In my heart I really want a clamshell grill (just a griddle with another plate that lowers on top of the food) but I suppose seeing the food change from pink to brown is kinda satisfying.

Can you spill the secrets of what part two is going to be? 👀 

Gosh this is all so exciting that I'm going to get up, get back into the kitchen and make some pizza.

**Comment by Qustinnus** [comment]
round 2 will probably be soup

**Comment by KathrinBailey** [comment]
[POV - You walk through service trying to find a ClothesMate](https://youtu.be/nVDRnDyZWDo)

**Comment by ilikesaying** [comment]
burger time

**Review by Twaticus — APPROVED** [review]
god YES

**Comment by RaveRadbury** [comment]
Pancakes are made on griddles. As are hashbrowns. And fried eggs. This is important.

**Comment by KathrinBailey** [comment]
> Pancakes are made on griddles. As are hashbrowns. And fried eggs. This is important.

Just use a pan, you get to choose the size and thickness of the pancake then. 👀

**Comment by Nickkoranda** [comment]
Add a griddle in the middle of the serving tables for a hibachi experience

**Comment by Qustinnus** [comment]
Pancakes etc will be done in a separate PR, I need to think about whether or not I'll add pans, I'm considering pots though.

**Comment by wesoda25** [comment]
What a brave new world we live in

**Comment by TheVekter** [comment]
Will griddles be buildable? I can only assume so.

**Comment by Misdoubtful** [comment]
You just, put the meat on it? Amazing.

**Comment by Owai-Seek** [comment]
Are you still going to be able to cook things on bonfires?

**Comment by Qustinnus** [comment]
Think so, I didn't touch it much but I'll do a double check.

**Comment by Qustinnus** [comment]
added the griddle to each map @EOBGames 
image
image
image
image

**Review by EOBGames — APPROVED** [review]
Approving the map changes, they all seem fine to me.

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+/obj/machinery/griddle/Initialize()
+	. = ..()
+	grill_loop = new(list(src), FALSE)
+	variant = rand(1,3)
+
+/obj/machinery/griddle/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(griddled_objects.len >= max_items)
+		to_chat(user, "<span class='notice'>[src] can't fit more items!</span>")
+		return
```
Comment (LemonInTheDark) [inline code/modules/food_and_drinks/kitchen_machinery/griddle.dm:34]: Is it possible to throw something onto the grill?

Comment (Qustinnus) [reply code/modules/food_and_drinks/kitchen_machinery/griddle.dm:34]: not at the moment, not sure how I could best add it

Comment (LemonInTheDark) [reply code/modules/food_and_drinks/kitchen_machinery/griddle.dm:34]: my inner idea guy has ideas, but they're not too important. I'm good with this.

**Comment by Shitpostmcgillygully** [comment]
This shit is so scuffed

Clicking on patty+bun with the other deletes both
Clicking on steak+bun with the other deletes both

There is no explanation as to how to make pattys or even that you need to except through the crafting menu which doesn't tell you how to even begin

Why on earth is this merged and upstream

**Comment by Qustinnus** [comment]
those two bugs you mentioned aren't even related to the griddle lol

**Comment by Ninija159** [comment]
How hard would it be, to turn the Energy sword(possibly Double one too) into a cooking/cutting tool like the kitchen knife? I'd love to see this added. because currently you can't slice bread or meat into cutlets with it, possibly Make it so it cooks the meat or maybe something else.

**Pull request merged**